opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java

```diff
@@ -595,12 +622,49 @@ public String getVersion() {
         return "2.0";
     }
 
-    public void setInputTransformer(InputTransformer inputTransformer) {
-        this.inputTransformer = inputTransformer;
+    private InputTransformer getInputTransformer(InputStream inputStream) {
+        XMLStreamReader xmlStreamReader = null;
+        try {
+            xmlStreamReader = xmlInputFactory.createXMLStreamReader(inputStream);
+            Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+            if (bundle == null) {
+                // only happens during unit tests
+                return defaultInputTransformer;
+            }
+            BundleContext bundleContext = bundle.getBundleContext();
```

Comment:
You could move the FrameworkUtils  call to a protected method and override in your unit tests so you can mock this behavior. I would much rather see that.